x-pack/plugins/remote_clusters/public/sections/remote_cluster_form/remote_cluster_form.js

```diff
@@ -0,0 +1,617 @@
+} from '../../services';
+
+const defaultFields = {
+  name: '',
+  seeds: [],
+  skipUnavailable: false,
+};
+
+export class RemoteClusterFormUi extends Component {
+  static propTypes = {
+    save: PropTypes.func.isRequired,
+    cancel: PropTypes.func,
+    isSaving: PropTypes.bool,
+    saveError: PropTypes.object,
+    fields: PropTypes.object,
+    disabledFields: PropTypes.object,
+  }
+
+  static defaultProps = {
+    fields: merge({}, defaultFields),
+    disabledFields: {},
+  }
+
+  constructor(props) {
+    super(props);
+
+    const { fields, disabledFields } = props;
+    const fieldsState = merge({}, defaultFields, fields);
+
+    this.state = {
+      localSeedErrors: [],
+      fields: fieldsState,
+      disabledFields,
+      fieldsErrors: this.getFieldsErrors(fieldsState),
+      areErrorsVisible: false,
+    };
+  }
+
+  getFieldsErrors(fields) {
+    const { name, seeds } = fields;
+
+    const errors = {};
+
+    if (!name || !name.trim()) {
+      errors.name = (
+        <FormattedMessage
+          id="xpack.remoteClusters.form.errors.nameMissing"
+          defaultMessage="Name is required."
+        />
+      );
+    } else if (name.match(/[^a-zA-Z\d\-_]/)) {
+      errors.name = (
+        <FormattedMessage
+          id="xpack.remoteClusters.form.errors.illegalCharacters"
+          defaultMessage="Name can only contain letters, numbers, underscores, and dashes."
+        />
+      );
+    }
+
+    if (!seeds.some(seed => Boolean(seed.trim()))) {
+      errors.seeds = (
+        <FormattedMessage
+          id="xpack.remoteClusters.form.errors.seedMissing"
+          defaultMessage="At least one seed is required."
+        />
+      );
+    }
+
+    return errors;
+  }
+
+  onFieldsChange = (changedFields) => {
+    const { fields: prevFields } = this.state;
+
+    const newFields = {
+      ...prevFields,
+      ...changedFields,
+    };
+
+    this.setState({
+      fields: newFields,
+      fieldsErrors: this.getFieldsErrors(newFields),
+      areErrorsVisible: false,
+    });
+  };
+
+  getAllFields() {
+    const {
+      fields: {
+        name,
+        seeds,
+        skipUnavailable,
+      },
+    } = this.state;
+
+    return {
+      name,
+      seeds,
+      skipUnavailable,
+    };
+  }
+
+  save = () => {
+    const { save } = this.props;
+    const { fieldsErrors } = this.state;
+
+    if (Object.keys(fieldsErrors).length > 0) {
+      this.setState({
+        areErrorsVisible: true,
+      });
+      return;
+    }
+
+    const cluster = this.getAllFields();
+
+    save(cluster);
+  };
+
+  getLocalSeedErrors = (seedNode) => {
+    const { intl } = this.props;
+
+    const errors = [];
+
+    const isInvalid = !isSeedNodeValid(seedNode);
+
+    if (isInvalid) {
+      errors.push(intl.formatMessage({
+        id: 'xpack.remoteClusters.remoteClusterForm.localSeedError.invalidCharactersMessage',
+        defaultMessage: `Seed nodes must consist of valid characters, with optional parts separated
```

Comment:
__question:__ won't you have undesired new line char in this string? Maybe you want to just concatenate two ordinary strings instead?

Reply:
It renders OK because the whitespace gets collapsed:

![image](https://user-images.githubusercontent.com/1238659/49838709-d9595c00-fd60-11e8-94f3-9cafd8144b4f.png)

Reply:
Ah, good then!